src/Workspaces/Core/Portable/FindSymbols/SymbolFinder.cs

```diff
@@ -245,10 +246,11 @@ public static IEnumerable<TSymbol> FindSimilarSymbols<TSymbol>(TSymbol symbol, C
 
                 // GetDocument will return null for locations in #load'ed trees. TODO:  Remove this check and add logic
                 // to fetch the #load'ed tree's Document once https://github.com/dotnet/roslyn/issues/5260 is fixed.
-                // TODO: the assert is also commented out because generated syntax trees won't have a document until
-                // https://github.com/dotnet/roslyn/issues/42823 is fixed
                 if (originalDocument == null)
+                {
+                    Debug.Assert(solution.Workspace.Kind == WorkspaceKind.Interactive || solution.Workspace.Kind == WorkspaceKind.MiscellaneousFiles);
```

Comment:
This assert was also commented out when the other ones were; the code was then moved and the assert deleted without the comment being deleted. This restores to the original pattern.